File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        
        words = s.split()
        
        left =0
        right = len(words)-1
        
        while left< right:
            
            temp = words[left]
            words[left] = words[right]
            words[right] = temp
            left+=1
            right-=1
        
        return " ".join(words).strip(" ")

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        
        String[] words = s.split("\\s+");
        
        int left = 0;
        int right = words.length-1;
        
        while(left < right) {
            String temp = words[left];
            words[left] = words[right];
            words[right] = temp;
            left++;
            right--;
        }
        
        return String.join(" ", words).trim();
        
    }
}